lib/api/multiObjectDelete.js

```diff
@@ -271,9 +272,27 @@ export function getObjMetadataAndDelete(authInfo, canonicalID, request,
             // TODO: update number of objects to reflect creation of
             // delete markers; Utapi must be updated to allow this for
             // 'deleteObject' metrics
+            let isDeleteMarker;
+            let deleteMarker;
+            // If trying to delete an object that does not exist (empty objMD)
+            // return DeleteMarkerVersionId equals the versionID generated and
+            // DeleteMarker equals true
+            if (!objMD) {
+                isDeleteMarker = true;
+                deleteMarker = versionIdUtils.encode(versionId);
```

Comment:
instead of re-encoding the version ID here, you can probably use entry.versionId (the encoded version ID provided by the user)